### PR TITLE
[RFC] signal_init: cleanup macOS workaround

### PR DESCRIPTION
**Pull request by aktau**
As discussed on #5243.

### CODE REVIEW

File: src/nvim/os/signal.c [hunk 1]
```diff
@@ -158,24 +158,18 @@ static void on_signal(SignalWatcher *handle, int signum, void *data)
...
-# endif
-
-  sigaddset(&mask, SIGCHLD);
-
-# if defined(__APPLE__) && !defined(__clang__) && !defined(__INTEL_COMPILER) \
-    && (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6))
-#  pragma GCC diagnostic pop
+  // macOS has type-unsafe inline functions and macro's in signal.h, see
+  // #5243. This will start warning when Apple fixes their headers.
+# if defined(__APPLE__) && defined(__GNUC__) && !defined(__clang__)
```
Comment (justinmk) [inline src/nvim/os/signal.c]: Just checking for Apple is good enough IMO and arguably communicates the intent more accurately, no?

Comment (aktau) [inline src/nvim/os/signal.c]: Yea, I was thinking about that. Perhaps it doesn't matter that Clang on macOS doesn't warn for this.

There's one possible edge case, if GCC ever decides to stop spewing warnings for macro's from system headers like Clang, we would be none the wiser.

Comment (justinmk) [inline src/nvim/os/signal.c]: > There's one possible edge case, if GCC ever decides to stop spewing warnings for macro's from system headers like Clang, we would be none the wiser.

How would

    # if defined(__APPLE__) && defined(__GNUC__) && !defined(__clang__)

differ from

    # if defined(__APPLE__)

in that case?

File: src/nvim/os/signal.c [hunk 2]
```diff
@@ -158,24 +158,18 @@ static void on_signal(SignalWatcher *handle, int signum, void *data)
...
+  sigaddset((int *) &mask, signo);
```
Comment (justinmk) [inline src/nvim/os/signal.c]: style guidelines changed to disallow space after a cast

**Comment by aktau** [comment]
Because of the ugliness of unblocking just SIGCHLD, I got to thinking. There's no reason why we shouldn't just unblock all signals at the start, so we have a clean slate. So I did just that. The code is cleaner now, too. Please check it out.

**Comment by justinmk** [comment]
Merged. Thank you @aktau !

**Pull request closed without merging**